# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/midi/MIDIPortChild.h
# Commit: cfba299e5067
# Full Hash: cfba299e5067a4eab3f3b982374b0770970a3401
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-04-12 03:57:01
# Regressor Bug: 1758468
# File Overlap Count: 1
# Description:
#   Bug 1758468 - Use stable, anonymous IDs for Web MIDI ports r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142550
# ==============================================================================

diff -r c0eb987d0f76 -r cfba299e5067 dom/midi/MIDIPortChild.h
--- a/dom/midi/MIDIPortChild.h	Mon Apr 11 17:43:08 2022 +0000
+++ b/dom/midi/MIDIPortChild.h	Mon Apr 11 17:43:09 2022 +0000
@@ -7,8 +7,8 @@
 #ifndef mozilla_dom_MIDIPortChild_h
 #define mozilla_dom_MIDIPortChild_h
 
+#include "mozilla/dom/MIDIPortInterface.h"
 #include "mozilla/dom/PMIDIPortChild.h"
-#include "mozilla/dom/MIDIPortInterface.h"
 
 namespace mozilla::dom {
 
@@ -33,6 +33,8 @@
 
   MIDIPortChild(const MIDIPortInfo& aPortInfo, bool aSysexEnabled,
                 MIDIPort* aPort);
+  nsresult GenerateStableId(const nsACString& aOrigin);
+  const nsString& StableId() { return mStableId; };
   // virtual void Shutdown() override;
   void SetActorAlive();
 
@@ -43,6 +45,7 @@
   // Pointer to the DOM object this actor represents. The actor cannot outlive
   // the DOM object.
   MIDIPort* mDOMPort;
+  nsString mStableId;
   bool mActorWasAlive;
 };
 }  // namespace mozilla::dom